judge/views/submission.py

```diff
@@ -286,6 +286,8 @@ def get_context_data(self, **kwargs):
         context = super(SubmissionsListBase, self).get_context_data(**kwargs)
         authenticated = self.request.user.is_authenticated
         context['dynamic_update'] = False
+        if self.in_contest:
+            context['dynamic_contest_id'] = self.contest.id
```

Comment:
```suggestion
        context['dynamic_contest_id'] = self.in_contest and self.contest.id
```

This is mostly so that the variable is always defined. `django-jinja` turns undefined variables into `None`, but in normal jinja2 this would cause the template to crash.